opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/_view_instrument_match.py

```diff
@@ -126,7 +126,7 @@ def collect(
         self,
         aggregation_temporality: AggregationTemporality,
         collection_start_nanos: int,
-    ) -> Sequence[DataPointT]:
+    ) -> Optional[Sequence[DataPointT]]:
```

Comment:
I feel like we should just return an empty sequence here and the caller can check handle the empty case instead of `None`

Reply:
The aggregations [`collect`](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py#L93) method already can return `None`. It is more consistent if every other `collect` method can also return `None`.

Reply:
I don't follow how it's more consistent. In one case its an `Optional[T]` and in the other it's  `Optional[Sequence[T]]`, i.e. you're already removing the None values.

It would let you remove the if check on L139 and the caller doesn't have to distinguish between the cases of `None` and `[]`. You also don't have to update any of the signatures this way.

Reply:
> I don't follow how it's more consistent. In one case its an `Optional[T]` and in the other it's `Optional[Sequence[T]]`, i.e. you're already removing the None values.

It is more consistent in the sense that when there are no new metric points for a collection cycle any `collect` method from either an `Aggregation`, `MetricReaderStorage`, `MeasurementConsumer` or `ViewInstrumentMatch` will return `None`.

> It would let you remove the if check on L139

I made it more concise :+1: 

> and the caller doesn't have to distinguish between the cases of `None` and `[]`.

Not really, the caller still would have to check [the result from `ViewInstrumentMatch.collect`](https://github.com/open-telemetry/opentelemetry-python/pull/3335/files#diff-1d98ed0d6a0a4255b1e2d5e17246bc84af5c271c9e6dc988138575154f906f25R159).